frontend/common/src/components/Pagination/Pagination.test.tsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { render, screen } from "@testing-library/react";
+import { renderHook, act } from "@testing-library/react-hooks";
+import React from "react";
+import Pagination from "./Pagination";
+import { DOTS, usePagination } from "./usePagination";
+import IntlContainer from "../../../../talentsearch/resources/js/components/IntlContainer";
+
+function renderPagination(
+  currentPage: number,
+  pageSize: number,
+  siblingCount: number,
+  totalCount: number,
+  handlePageChange: (pageNumber: number) => void,
+) {
+  return render(
+    <IntlContainer locale="en">
+      <Pagination
+        currentPage={currentPage}
+        pageSize={pageSize}
+        siblingCount={siblingCount}
+        totalCount={totalCount}
+        handlePageChange={handlePageChange}
+      />
+    </IntlContainer>,
+  );
+}
+
+describe("Pagination tests", () => {
+  test("If the total page count is less then page pills show range from 1 to totalPageCount", () => {
+    const props = { currentPage: 1, pageSize: 2, siblingCount: 1, totalCount: 10 };
```

Comment:
Should we consider using same arg data from stories in tests? I think this ability is considered one of the perks of using storybook's [new Component Story Format (CSF)](https://storybook.js.org/docs/react/api/csf)

> Stories are a practical starting point for UI testing. Import stories into tools like Jest, Testing Library, Puppeteer, Cypress, and Playwright to save time and maintenance work.

https://storybook.js.org/docs/react/writing-tests/importing-stories-in-tests

Reply:
To be explicit with suggestion, just something like this (though RightDots isn't the same data here, so would need to be aligned):

```suggestion
    const props = RightDots.args;
```